dvc/remote/base.py

```diff
@@ -697,17 +697,18 @@ def checksum_to_path_info(self, checksum):
     def list_cache_paths(self, prefix=None, progress_callback=None):
```

Comment:
This is pretty much a `walk_files` method, but it is not supported in every remote yet, so we are using a legacy implementation for now.

Reply:
Using `walk_files` would've been much more intuitive, esp since `walk` presumes that it is an iterator, unlike `list_cache_paths` which could be anything.